datafusion/physical-expr/src/expressions/binary.rs

```diff
@@ -218,21 +132,6 @@ macro_rules! compute_utf8_op {
     }};
 }
 
-/// Invoke a compute kernel on a pair of binary data arrays
-macro_rules! compute_binary_op {
-    ($LEFT:expr, $RIGHT:expr, $OP:ident, $DT:ident) => {{
-        let ll = $LEFT
-            .as_any()
-            .downcast_ref::<$DT>()
-            .expect("compute_op failed to downcast left side array");
-        let rr = $RIGHT
-            .as_any()
-            .downcast_ref::<$DT>()
-            .expect("compute_op failed to downcast right side array");
-        Ok(Arc::new(paste::expr! {[<$OP _binary>]}(&ll, &rr)?))
-    }};
-}
-
 /// Invoke a compute kernel on a data array and a scalar value
 macro_rules! compute_utf8_op_scalar {
```

Comment:
it seems like eventually we could make a `ColumnarValue::to_arrow_datum` type function that would allow direct invocation of many of these kernels from DataFusion without needing layers of dispatch

Reply:
Yes, ideally with #7353 we could make `ScalarValue` implement `Datum` directly

Reply:
And then implement `Datum` directly for `ColumnarValue`  as well 🤔  that would be pretty sweet